float64[] q_array
float64[] tau_comp